File: blacksmith-of-lenai-app/src/app/shared/models/quest-data.ts
export interface QuestData {
  questId: string;
  questName: string;
  questDescription: string;
  isMainQuest: boolean;
  questDetails: QuestDetailData[];
  questRewards: QuestRewardData[];
}

export interface QuestDetailData {
  questDetailId: string;
  questDetailName: string;
}

export interface QuestRewardData {
  experience?: number;
  money?: number;
}

File: blacksmith-of-lenai-api/Models/QuestData.cs
namespace PlayFabApi.Models;

public sealed record QuestData
{
    public required Guid QuestId { get; init; }
    public required string QuestName { get; init; }
    public required bool IsMainQuest { get; init; }
    public required string QuestDescription { get; init; }
    public required List<QuestDetail> QuestDetails { get; init; }
    public required List<QuestReward> QuestRewards { get; init; }
}